src/merkle_sum_tree/mod.rs

```diff
@@ -1,22 +1,30 @@
 mod entry;
 mod mst;
+mod params;
 mod tests;
 mod utils;
 use halo2_proofs::halo2curves::bn256::Fr as Fp;
 
-#[derive(Default, Clone, Debug)]
-pub struct MerkleProof {
+pub use params::MST_WIDTH;
+pub use params::N_ASSETS;
+pub use params::R_L_ENTRY;
+pub use params::R_L_NODE;
+pub use params::WIDTH_ENTRY;
+pub use params::WIDTH_NODE;
+
+#[derive(Clone, Debug)]
+pub struct MerkleProof<const N_ASSETS: usize> {
     pub root_hash: Fp,
-    pub entry: Entry,
+    pub entry: Entry<N_ASSETS>,
     pub sibling_hashes: Vec<Fp>,
-    pub sibling_sums: Vec<Fp>,
+    pub sibling_sums: Vec<[Fp; N_ASSETS]>,
     pub path_indices: Vec<Fp>,
 }
```

Comment:
Related to the whole topic of switching from vectors to array, I'm saying that because I hope it can improve the performance of the `build_merkle_tree()`. If you are applying these changes, please benchmark it.

Reply:
Created #36 for this.